fixed in 5.12.5 by commit https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/scripts/kconfig/nconf.c?h=linux-5.12.y&id=3c96f542bf751c3e6c16c28b57d737aedbcc106e

so this patch is no longer needed for 5.12.5+ kernels!

diff --git a/scripts/kconfig/nconf.c b/scripts/kconfig/nconf.c
index ac92c0ded6c5..9f1d85acc572 100644
--- a/scripts/kconfig/nconf.c
+++ b/scripts/kconfig/nconf.c
@@ -505,7 +505,7 @@ static int get_mext_match(const char *match_str, match_f flag)
 
 	index = match_start;
 	index = (index + items_num) % items_num;
-	while (true) {
+	while (match_start != -1) {
 		char *str = k_menu_items[index].str;
 		if (strcasestr(str, match_str) != NULL)
 			return index;
@@ -516,8 +516,9 @@ static int get_mext_match(const char *match_str, match_f flag)
 			++index;
 		index = (index + items_num) % items_num;
 		if (index == match_start)
-			return -1;
+			break;
 	}
+	return -1;
 }
 
 /* Make a new item. */
